src/BloomBrowserUI/pageChooser/js/page-chooser.js

```diff
@@ -189,10 +189,13 @@ var PageChooser = (function () {
                 indexToSelect = index;
             var pageDescription = $(".pageDescription", div).first().text();
             $(".pageDescription", currentGridItemHtml).first().text(pageDescription);
-            var pageLabel = $(".pageLabel", div).first().text();
+            var pageLabel = $(".pageLabel", div).first().text().trim();
             $(".gridItemCaption", currentGridItemHtml).first().text(pageLabel);
+            // any changes to how we tweak the page label to get a file name
+            // must also be made in EnhancedImageServer.FindOrGenerateImage().
             pageLabel = pageLabel.replace("&", "+"); //ampersands don't work in the svg file names, so we use "+" instead
-            $("img", currentGridItemHtml).attr("src", pageFolderUrl + "/template" + "/" + pageLabel + ".svg");
+            // gensvg is a 'magic' extension which the Bloom fileserver understands. See EnhancedImageServer.FindOrGenerateImage.
+            $("img", currentGridItemHtml).attr("src", pageFolderUrl + "/template" + "/" + pageLabel + ".gensvg");
```

Comment:
Perhaps more idiomatic for web stuff would be to instead add a query, so you'd have;

   pageLabel + ".svg?generateThumbnaiIfNecessary=true"